sway-lib-std/src/tx.sw

```diff
@@ -186,12 +195,18 @@ pub fn tx_input_type(ptr: u32) -> u8 {
     }
 }
 
-/// If the input's type is `InputCoin`, return the owner.
-/// Otherwise, undefined behavior.
-pub fn tx_input_coin_owner(input_ptr: u32) -> Address {
-    let owner_addr = ~Address::from(asm(buffer, ptr: input_ptr) {
-        // Need to skip over six words, so add 8*6=48
-        addi ptr ptr i48;
+/// Get the type of an input at a given index
+fn get_input_type(index: u8) -> u8 {
+    let ptr = tx_input_pointer(index);
+    let input_type = tx_input_type(ptr);
+    input_type
+}
+
+/// Read 256 bits from memory at a given offset from a given pointer
+fn get_b256_from_pointer_offset(pointer: u32, offset: u32) -> b256 {
```

Comment:
In order to use this function to replace the current `tx_input_coin_owner()`/`tx_input_message_owner()`, one would need to know:
- the input pointer (already a requirement with the current model), &
- the offset. 
So, we would need to make the offsets available as consts in the stdlib or something.
The current approach I've taken is to create a `tx_input_owner()` function which returns a result, using the `Err` variant for inputs other than `coin` or `message`.
I'm not sure if these offsets will be changing with the introduction of the `gtf` instruction as well?
cc @adlerjohn

Reply:
Ah, I should have scrolled down more; I see that you've simply wrapped this function with `get_input_coin_owner()` avoiding the need to have users manually pass in offsets.

Reply:
In general, getting a `b256` value is a common enough operation to warrant this function IMO.